xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/specifiers/LedgerSpecifier.java

```diff
@@ -0,0 +1,161 @@
+package org.xrpl.xrpl4j.model.client.specifiers;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Preconditions;
+import org.immutables.value.Value;
+import org.xrpl.xrpl4j.model.jackson.modules.LedgerSpecifierDeserializer;
+import org.xrpl.xrpl4j.model.transactions.Hash256;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * Represents one of the three ways of specifying a ledger in a rippled API request.
+ *
+ * @see "https://xrpl.org/basic-data-types.html#specifying-ledgers"
+ */
+@Value.Immutable
+@JsonSerialize(as = ImmutableLedgerSpecifier.class)
+@JsonDeserialize(as = ImmutableLedgerSpecifier.class, using = LedgerSpecifierDeserializer.class)
+public interface LedgerSpecifier {
+
+  /**
+   * Construct a {@link LedgerSpecifier} with a ledger hash.
+   *
+   * @param ledgerHash A {@link Hash256} containing the ledger hash of the desired ledger.
+   *
+   * @return A {@link LedgerSpecifier} containing {@code ledgerHash}.
+   */
+  static LedgerSpecifier ledgerHash(Hash256 ledgerHash) {
+    return ImmutableLedgerSpecifier.builder()
```

Comment:
Or these? I call all of these methods in tests...

Reply:
Showing as tested in the latest report.